slides/aml-10-021918-trees-forests/index.html

```diff
@@ -0,0 +1,724 @@
+      }
+      .normal {
+          font-size: 30px;
+      }
+      .quote_author {
+          display: block;
+          text-align: right;
+          margin-top: 20px;
+          font-size: 30px;
+          font-family: 'Garamond';
+      }
+      .larger, .larger .remark-code {
+          font-size: 40px;
+      }
+      .largest, .largest .remark-code {
+          font-size: 50px;
+      }
+      .left-column, .right-column {
+          width: 48%;
+      }
+      .right-column{
+          float: right;
+      }
+      .left-column{
+          float: left;
+      }
+      .narrow-right-column {
+          float: right;
+          width: 32%
+      }
+      .wide-left-column {
+          float: left;
+          width: 65%
+      }
+      .invisible {
+          visibility: hidden
+      }
+      .remark-code, .remark-inline-code { font-family: 'Ubuntu Mono'; }
+
+      .column:first-of-type {float:left}
+      .column:last-of-type {float:right}
+      .split-40 .column:first-of-type {width: 50%}
+      .split-40 .column:last-of-type {width: 50%}
+    </style>
+  </head>
+  <body>
+    <textarea id="source">
+
+class: center, middle
+
+### W4995 Applied Machine Learning
+
+# Trees &amp; Forests
+
+02/14/18
+
+Andreas C. Müller
+
+???
+N/A
+
+---
+class: spacious
+# Why Trees?
+- Very powerful modeling method – non-linear!
+- Doesn’t care about scaling of distribution of data!
+- “Interpretable”
+- Basis of very powerful models!
+???
+---
+
+class: centre,middle
+# Decision Trees for Classification
+???
+---
+
+# Idea: series of binary questions
+.center[
+![:scale 70%](images/img_1.png)
+]
+???
+
+---
+class: split-40
+# Building Trees
+
+.column[
+![:scale 70%](images/img_2.png)
+ - Continuous features:
+ - “questions” are thresholds on single features.
+ - [Other methods are possible but not as common]
+ - For each split: exhaustive search over all features and thresholds!
+ - Minimize “impurity”
+]
+.column[
+![:scale 50%](images/img_3.png)
+---
+![:scale 50%](images/img_4.png)
+]
+
+???
+Does the layout look okay??
+---
+
+# Criteria (for classification)
+ - Gini Index: 
+   --
+   .center[![:scale 50%](images/img_5.png)]
+ - Cross-Entropy:
+   --
+   .center[![:scale 50%](images/img_6.png)]
+
+
+ 
+ ![:scale 5%](images/img_7.png) observations in node m
+ 
+ ![:scale 3%](images/img_8.png) classes
+ 
+ ![:scale 5%](images/img_9.png) distribution over classes in node m
+???
+---
+
+# Prediction
+
+.center[
+![:scale 80%](images/img_10.png)
+]
+
+ - Traverse tree based on feature tests
+ - Predict most common class in leaf
+
+???
+
+---
+
+# Regression trees
+
+ - Impurity Criteria:
+ - Mean Squared Error
+ - Mean Absolute Error
+ 
+ 
+ - Prediction:
+ - Predict mean.
+ 
+
+ - Without regularization / pruning:
+ - Each leaf often contains a single point to be “pure”
+???
+
+---
+
+# Visualizing trees with sklearn
+
+.smaller[
+```python
+from sklearn.datasets import load_breast_cancer
+cancer = load_breast_cancer()
```

Comment:
If there is no output in between you can make it one big code block.

Reply:
Resolved